presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxRecordSetProvider.java

```diff
@@ -62,116 +67,154 @@ public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorS
         requireNonNull(columns, "columns is null");
         checkArgument(!columns.isEmpty(), "must provide at least one column");
 
+        ImmutableMap<String, Type> columnTypes = getColumnTypes(columns);
```

Comment:
Yeah, I talked to @raghavsethi after I made all those comments. I think relying on a map when ordering matters is not the right thing to do. In that case, that should be made explicit by using a List (of pairs or entries)